### PR TITLE
Fix/bump core

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**

### CODE REVIEW

File: utils/vamm-math/VAMMBase.sol [hunk 1]
```diff
@@ -154,6 +154,27 @@ library VAMMBase {
         ).unwrap();
     }
```
Comment (ioanaroceanu2000) [inline utils/vamm-math/VAMMBase.sol:156]: @arturbeg pls review

File: src/storage/DatedIrsVamm.sol [hunk 2]
```diff
@@ -747,29 +783,46 @@ library DatedIrsVamm {
...
+        ).toUint();
 
         // Compute unfilled tokens in our range and to the right of the current tick
         unfilledBaseTokensRight = self.baseBetweenTicks(
             tickLower > self.vars.tick ? tickLower : self.vars.tick, // max(tickLower, currentTick)
             tickUpper > self.vars.tick ? tickUpper : self.vars.tick,  // max(tickUpper, currentTick)
             liquidityPerTick.toInt()
         ).toUint();
+
+        uint256 secondsTillMaturity = self.immutableConfig.maturityTimestamp - block.timestamp;
```
Comment (ioanaroceanu2000) [inline src/storage/DatedIrsVamm.sol:812]: @arturbeg pls review

File: src/storage/DatedIrsVamm.sol [hunk 3]
```diff
@@ -750,26 +782,106 @@ library DatedIrsVamm {
...
+            leftLowerTick,
+            leftUpperTick,
+            liquidityPerTick
         ).toUint();
+
+        if ( unfilledBaseTokensLeft == 0 ) {
+            return (0, 0);
+        }
+
+        // unfilledBaseTokensLeft is negative
```
Comment (ioanaroceanu2000) [inline src/storage/DatedIrsVamm.sol:831]: @0xZenus

File: src/storage/DatedIrsVamm.sol [hunk 4]
```diff
@@ -690,32 +690,64 @@ library DatedIrsVamm {
...
+                    );
                 unfilledBaseLong += unfilledLongBase;
                 unfilledBaseShort += unfilledShortBase;
+                unfilledQuoteLong += unfilledLongQuote;
+                unfilledQuoteShort += unfilledShortQuote;
             }
         }
     }
 
+    function _getUnfilledBalancesFromPosition(
```
Comment (0xZenus) [inline src/storage/DatedIrsVamm.sol:728]: Let's avoid putting underline before function names. It decreases readability.

File: src/storage/DatedIrsVamm.sol [hunk 5]
```diff
@@ -750,26 +782,106 @@ library DatedIrsVamm {
...
+        (unfilledBaseTokensRight, unfilledQuoteTokensRight) = self._getUnfilledBalancesRight(
             tickLower > self.vars.tick ? tickLower : self.vars.tick, // max(tickLower, currentTick)
             tickUpper > self.vars.tick ? tickUpper : self.vars.tick,  // max(tickUpper, currentTick)
-            liquidityPerTick.toInt()
+            liquidityPerTick.toInt(),
+            secondsTillMaturity
+        );
+    }
+
+    function _getUnfilledBalancesLeft(
```
Comment (0xZenus) [inline src/storage/DatedIrsVamm.sol:811]: Same as above.

**Review by 0xZenus — APPROVED** [review]

**Pull request merged**